### PR TITLE
feat(core/xref): use inline references to provide context

### PR DESCRIPTION
**Pull request by sidvishnoi**
Uses nearby `[[SPEC]]` to provide context as `specs` to search for (as a part of https://github.com/w3c/respec/issues/1662)

Maybe this should be enabled via a flag, not always. Why? Consider:
``` html
<section data-cite="html">Uses [[WEBIDL]] to create context for <a>object</a></section>
```
Above, the query for this term will contain `specs: ["html", "webidl"]`, which results in an ambiguity that can't be resolved unless (either):
- User removes `data-cite="html"` from parent or remove the nearby `[[WEBIDL]]` 
- Instead of `specs.push(...inlineContextSpecs)`, we do `specs = inlineContextSpecs`

### CODE REVIEW

**Comment by marcoscaceres** [comment]
So, given:

```
<section data-cite="html">Uses [[WEBIDL]] to create context for <a>object</a></section>
```

I'm not sure I agree... that's clearly ambiguous, and the Editor should deal with.

**Comment by marcoscaceres** [comment]
updated example above.

**Comment by sidvishnoi** [comment]
So, if the element itself has `data-cite`, it wins. No inline context need to be taken into account. Right?

**Comment by marcoscaceres** [comment]
> So, if the element itself has data-cite, it wins. No inline context need to be taken into account. Right?

Correct. `.closest()` wins.

File: src/core/xref.js [hunk 1]
```diff
@@ -46,6 +46,17 @@ function createXrefMap(elems) {
           .replace(/!/g, "")
           .split(" ")
       : [];
+    if (datacite !== elem) {
+      // if element itself contains data-cite, we don't take inline context into account
+      const inlineContext = elem.parentElement.querySelectorAll("a.bibref");
```
Comment (marcoscaceres) [inline src/core/xref.js]: discussed making this search wider... up to `section` level .

File: src/core/xref.js [hunk 2]
```diff
@@ -46,6 +46,17 @@ function createXrefMap(elems) {
           .replace(/!/g, "")
           .split(" ")
       : [];
+    if (datacite !== elem) {
+      // if element itself contains data-cite, we don't take inline context into account
+      const refs = elem.closest("section").querySelectorAll("a.bibref");
+      if (refs) {
+        const inlineContextSpecs = [...refs].map(el =>
+          el.textContent.toLowerCase().replace(/^!/, "")
```
Comment (marcoscaceres) [inline src/core/xref.js]: We should do the removal of the “!” when the inline references are converted. So, in core/inlines i think.

File: src/core/xref.js [hunk 3]
```diff
@@ -46,6 +46,14 @@ function createXrefMap(elems) {
           .replace(/!/g, "")
           .split(" ")
       : [];
+    if (datacite !== elem) {
+      // if element itself contains data-cite, we don't take inline context into account
+      const refs = elem.closest("section").querySelectorAll("a.bibref");
```
Comment (marcoscaceres) [inline src/core/xref.js]: might be cleaner to just do:
```
const refs = [...elem.closest("section").querySelectorAll("a.bibref")]
  .map(el => el.textContent.toLowerCase());
specs.push(...refs);
```

File: src/core/xref.js [hunk 4]
```diff
@@ -46,6 +46,14 @@ function createXrefMap(elems) {
...
+      if (refs) {
```
Comment (marcoscaceres) [inline src/core/xref.js]: note that `refs` is always true, because ref is an array object.

**Pull request merged**